src/main/java/address/controller/PersonOverviewController.java

```diff
@@ -144,6 +153,12 @@ private void handleNewPerson() {
      */
     private void handleRetagPersons() {
         List<ReadOnlyViewablePerson> selectedPersons = personListView.getSelectionModel().getSelectedItems();
+
+        if (selectedPersons.isEmpty() || isListContainingNullObject(selectedPersons)) {
```

Comment:
This gets repeated many times, i suggest refactoring it out as a separate method, then can remove the isListContainingNullObject method too, something maybe like:
```java
public static boolean isSelectionValid() {
    return personListView.getSelectionModel().getSelectedItems().stream().anyMatch(Objects::isNull);
}
```